src/aspire/image/image_stacker.py

```diff
@@ -0,0 +1,291 @@
+
+        :return: 2D Numpy array.
+        """
+
+        # Careful, we need to reset this in case this stack is
+        # different from last call.
+        self._return_image_size = None
+
+        # Flatten image data.
+        if isinstance(stack, Image):
+            # Store the image size for returning later.
+            self._return_image_size = (1,) + stack.shape[1:]
+
+            # Sanity check dimensions.
+            if stack.stack_ndim != 1:
+                raise ValueError(
+                    f"`stack` shape of Image should be 1D for ImageStacking not {stack.stack_shape}."
+                    "  Try Image.stack_reshape if needed."
+                )
+            stack = stack.asnumpy().reshape(stack.n_images, -1)
+
+        # By this point we should always be an array with signal data
+        # in the last axis.
+        if not isinstance(stack, np.ndarray):
+            raise ValueError("`stack` should be `Image` instance or Numpy array.")
+        elif stack.ndim != 2:
+            raise ValueError(
+                f"`stack` numpy array shape should be 2D for ImageStacker not {stack.shape}."
+                "  Try Image.stack_reshape if needed."
+            )
+
+        return stack
+
+    def _return(self, result):
+        """
+        When ImageStacker has been passed an `Image`,
+        this method creates an `Image` instance to be returned.
+
+        In the case we started with a Numpy array this should pass
+        through.
+
+        :param result: Result data as Numpy Array.
+
+        :return: Image(result) or Numpy array(result) based on initial `stack` input type.
+        """
+        if self._return_image_size is not None:
+            result = Image(result.reshape(self._return_image_size))
+        return result
+
+
+class MeanImageStacker(ImageStacker):
+    """
+    Stack using `mean`.
+    """
+
+    def __call__(self, stack):
+        stack = self._check_and_convert(stack)
+
+        return self._return(stack.mean(axis=0))
+
+
+class MedianImageStacker(ImageStacker):
+    """
+    Stack using `median`.
+    """
+
+    def __call__(self, stack):
+        stack = self._check_and_convert(stack)
+
+        return self._return(np.median(stack, axis=0))
+
+
+class SigmaRejectionImageStacker(ImageStacker):
+    """
+    Stack using Sigma Rejection.
+
+    When no outliers exist, sigma rejection is equivalent to `mean`.
+    In the presence of outliers, pixels outside of `rejection_sigma`
+    from the per-pixel-mean are discarded before the mean is
+    performed.
+
+    For potentially less Gaussian distributions, 'FWHM' and 'FWTM'
+    methods are also provided.  These will take the mean of all values
+    laying above half the maximum and tenth maximum
+    respectively. Essentially this is the same wing clipping
+    procedure, but the location of the clip is determined by the peak
+    intensity instead of standard deviation.
+
+    Note, in both cases, user's are responsible for ensuring methods
```

Comment:
users

Reply:
yeah, i played with that sentence too much 😇 . fixed thanks